src/Common/ZooKeeper/ZooKeeper.cpp

```diff
@@ -55,6 +62,62 @@ static void check(Coordination::Error code, const std::string & path)
         throw KeeperException::fromPath(code, path);
 }
 
+ZooKeeperAvailabilityZoneMap& ZooKeeperAvailabilityZoneMap::instance()
+{
+    static ZooKeeperAvailabilityZoneMap map;
+    return map;
+}
+
+std::string ZooKeeperAvailabilityZoneMap::get(const std::string &host)
+{
+    std::lock_guard lock(mutex);
+    auto it = az_by_host.find(host);
+    if (it == az_by_host.end())
+        throw DB::Exception(DB::ErrorCodes::BAD_ARGUMENTS,"No availability zone info for host {}", host);
+    return it->second;
+}
+
+void ZooKeeperAvailabilityZoneMap::update(const std::string &host, const std::string &availability_zone)
+{
+    std::lock_guard lock(mutex);
+    az_by_host[host] = availability_zone;
+}
+
+std::vector<std::string> ZooKeeperAvailabilityZoneMap::shuffleHosts(const std::string & local_az, const std::vector<std::string> & hosts)
```

Comment:
Return string is not enough. ZooKeeperImpl requires original index in the https://github.com/ClickHouse/ClickHouse/blob/136efba321af48c985b17d97d4108ea00aa56841/tests/integration/test_zookeeper_fallback_session/test.py#L62

Reply:
fixed.